target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl838x.c

```diff
@@ -1891,27 +1891,27 @@ int rtl838x_write_phy(u32 port, u32 page, u32 reg, u32 val)
 int rtl838x_read_mmd_phy(u32 port, u32 addr, u32 reg, u32 *val)
 {
 	u32 v;
+	u32 park_page;
 
 	mutex_lock(&smi_lock);
 
-	if (rtl838x_smi_wait_op(100000))
-		goto timeout;
-
-	sw_w32(1 << port, RTL838X_SMI_ACCESS_PHY_CTRL_0);
-	mdelay(10);
-
-	sw_w32_mask(0xffff0000, port << 16, RTL838X_SMI_ACCESS_PHY_CTRL_2);
+	reg = mdiobus_c45_regad(reg);
+	pr_debug("%s port %d, addr %d, reg %d\n", __func__, port, addr, reg);
+	sw_w32(port << 16, RTL838X_SMI_ACCESS_PHY_CTRL_2);
 
 	v = addr << 16 | reg;
 	sw_w32(v, RTL838X_SMI_ACCESS_PHY_CTRL_3);
 
-	/* mmd-access | read | cmd-start */
-	v = 1 << 1 | 0 << 2 | 1;
-	sw_w32(v, RTL838X_SMI_ACCESS_PHY_CTRL_1);
+	park_page = sw_r32(RTL838X_SMI_ACCESS_PHY_CTRL_1) & ((0x1f << 15) | 0x2);
```

Comment:
This extracts the PARK_PAGE field _and_ hard sets the TYPE field (to MMD). AFAICT PARK_PAGE isn't actually set anywhere at all, so does this just depend on the last value it was set to before this driver runs?

Do you know what value is stored here? According to the SDK, a value of 0x1f means something along the lines of "ignore parked page". I have no idea what a "parked page" is however, but even the SDK seems to almost always use 0x1f. I've found a few occurrences of `PARK_PAGE = 0`

Reply:
the PARK_PAGE MR is now #11090 and we can move the discussion there.